src/main/java/com/systemmeltdown/meltdownlog/topics/DoubleTopic.java

```diff
@@ -0,0 +1,38 @@
+package com.systemmeltdown.meltdownlog.topics;
+
+import com.systemmeltdown.meltdownlog.lib.Utils;
+
+public class DoubleTopic extends LogTopic {
+	private double m_roundingFactor;
+	private double m_lastRoundedValue = Double.MIN_VALUE;
+	private long m_lastNanos = Long.MIN_VALUE;
+
+	public DoubleTopic(String name, double roundingFactor) {
+		super(name, Double.class);
+		m_roundingFactor = roundingFactor;
+	}
+
+	public void log(double value) {
+		log(value, System.nanoTime());
+	}
+
+	public void log(double value, long nanos) {
+		final double roundedValue = Utils.roundByFactor(value, m_roundingFactor);
+		if (roundedValue == m_lastRoundedValue) {
```

Comment:
Same issue here where the first value might not be written to the log.

Reply:
I have now consolidated this functionality in a single class, DataTopic.
Thanks!